common/include/edgeai_msc.h

```diff
@@ -0,0 +1,96 @@
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef _TI_EDGEAI_TIOVX_MSC_H_
+#define _TI_EDGEAI_TIOVX_MSC_H_
+
+/* Modules Headers */
+#include <tiovx_multi_scaler_module.h>
+#include <common/include/edgeai_pre_proc.h>
+
+namespace ti::edgeai::common
+{
+    using namespace std;
+
+    /**
+     * \brief Class that wraps the multiscaler related configuration
+     *
+     * \ingroup group_edgeai_common
+     */
+
+    class multiScaler
+    {
+        public:
+            /* Default constructor. Use the compiler generated default one. */
+            multiScaler();
+
+            /* Destructor. */
+            ~multiScaler();
+
+            /**
+             * Helper function to dump the configuration information.
+             */
+            void        dumpInfo();
+
+            /** Helper function to parse MSC configuration.
+             *
+             * @param input_wd Width of the input frame
+             * @param input_ht Height of the input frame
+             * @param post_proc_wd Width of the output frame (for post-process)
+             * @param post_proc_ht Height of the output frame (for post-process)
+             * @param pre_proc_obj Pre-Processing object
+             *
+             */
+            int32_t     getConfig(int32_t input_wd,
+                                  int32_t input_ht,
+                                  int32_t post_proc_wd,
+                                  int32_t post_proc_ht,
+                                  preProc *pre_proc_obj);
+
+        public:
+            /* Data structure passed to MSC module */
+            TIOVXMultiScalerModuleObj     multiScalerObj1;
+            
+            /* Optional second MSC to be used when need to scale by a factor
+             * more than 4.
+             */
+            TIOVXMultiScalerModuleObj     multiScalerObj2;
```

Comment:
Why not make multiScalerObj as a vector/array instead of putting multiple ones (1,2,etc)?
Also, for AHP, there are 2 VPACs, how is that accounted for here?

Reply:
I think this is more understandable. It can be clearly understood as to why we have two multiscaler objects and what is their utility. I have not addressed 2 VPAC issue (for AHP) for now, as I would like to spend some time on it to do it in such a way that it uses the functionality of both VPACs.